clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java

```diff
@@ -122,6 +123,48 @@ public void setup() {
         this.props.put(VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
     }
 
+    @Test
+    public void testOffsetFetchRequestStateToStringBase() {
+        ConsumerConfig config = mock(ConsumerConfig.class);
+        CommitRequestManager.MemberInfo memberInfo = new CommitRequestManager.MemberInfo();
+
+        CommitRequestManager commitRequestManager = new CommitRequestManager(
+                time,
+                logContext,
+                subscriptionState,
+                config,
+                coordinatorRequestManager,
+                offsetCommitCallbackInvoker,
+                "groupId",
+                Optional.of("groupInstanceId"),
+                metrics);
+
+        commitRequestManager.onMemberEpochUpdated(Optional.of(1), Optional.empty());
+        Set<TopicPartition> requestedPartitions = Collections.singleton(new TopicPartition("topic-1", 1));
+
+        CommitRequestManager.OffsetFetchRequestState offsetFetchRequestState = commitRequestManager.new OffsetFetchRequestState(
+                requestedPartitions,
+                retryBackoffMs,
+                retryBackoffMaxMs,
+                1000,
+                memberInfo);
```

Comment:
in the line#142 - The `memberInfo`  updated by `onMemberEpochUpdated` is the inner variable of `commitRequestManager`. Hence, this `memberInfo` used to create `OffsetFetchRequestState` is NOT updated by `onMemberEpochUpdated`

If we want to complete the test, maybe we can remove `MemberInfo` from constructor of `OffsetFetchRequestState`. After all, both `RetriableRequestState` and `OffsetFetchRequestState` are the inner class of `CommitRequestManager`, so it is valid to share the `MemberInfo` between those classes.

Reply:
I am a bit hesitant to change the constructor of ```OffsetFetchRequestState``` as I do not want to mess up usages elsewhere in the codebase. I did however find a way to achieve updating the inner ```memberInfo```. Test is working as well, I just needed to change the method from private to protected. Let me know what you think